hack/tools/go.mod

```diff
@@ -6,20 +6,19 @@ require (
 	github.com/joelanford/go-apidiff v0.6.0
 	github.com/onsi/ginkgo/v2 v2.11.0
 	k8s.io/code-generator v0.27.3
-	sigs.k8s.io/cluster-api/hack/tools v0.0.0-20230711141913-5dedc1b4f9d5
+	sigs.k8s.io/cluster-api/hack/tools v0.0.0-20230720182410-9c0a017bd4e3
```

Comment:
@chrischdi Let's make sure we bump this to the v1.5.0 commit next week. But I think for now it's fine to change this from the rc.0 commit to the latest commit of the release-1.5 release branch